Ambiguous overload for ‘==‘ between C Vk* and C++ vk::* handles when using spaceship operator

Observed under C++20 and above. Setting `VULKAN_HPP_NO_SPACESHIP_OPERATOR=ON` prevents this issue, so it has something to do with the `<=>` operator. 

Compiler logs:
<details><summary>GCC 15</summary>

```cpp
Vulkan-Hpp/tests/Handles/Handles.cpp:60:13: error: ambiguous overload for ‘operator==’ (operand types are ‘VkInstance’ {aka ‘VkInstance_T*’} and ‘vk::Instance’)
   60 |     if (vki == instance)
      |         ~~~ ^~ ~~~~~~~~
      |         |      |
      |         |      vk::Instance
      |         VkInstance {aka VkInstance_T*}
Vulkan-Hpp/tests/Handles/Handles.cpp:60:13: note: there are 7 candidates
   60 |     if (vki == instance)
      |         ~~~~^~~~~~~~~~~
In file included from Vulkan-Hpp/vulkan/vulkan.hpp:9408,
                 from Vulkan-Hpp/vulkan/vulkan_raii.hpp:11,
                 from Vulkan-Hpp/tests/Handles/Handles.cpp:22:
Vulkan-Hpp/vulkan/vulkan_handles.hpp:21233:10: note: candidate 1: ‘constexpr bool vk::Instance::operator==(const vk::Instance&) const’ (reversed)
21233 |     auto operator<=>( Instance const & ) const = default;
      |          ^~~~~~~~
Vulkan-Hpp/tests/Handles/Handles.cpp:60:13: note: candidate 2: ‘operator==(VkInstance {aka VkInstance_T*}, VkInstance {aka VkInstance_T*})’ (built-in)
   60 |     if (vki == instance)
      |         ~~~~^~~~~~~~~~~
```

</details>

<details><summary>Clang 20</summary>

```cpp
Vulkan-Hpp/tests/Handles/Handles.cpp:60:13: error: use of overloaded operator '==' is ambiguous (with operand types 'VkInstance' (aka 'VkInstance_T *') and 'vk::Instance')
   60 |     if (vki == instance)
      |         ~~~ ^  ~~~~~~~~
Vulkan-Hpp/tests/Handles/Handles.cpp:60:13: note: built-in candidate operator==(struct VkInstance_T *, struct VkInstance_T *)
Vulkan-Hpp/tests/Handles/Handles.cpp:60:13: note: built-in candidate operator==(const struct VkInstance_T *, const struct VkInstance_T *)
Vulkan-Hpp/tests/Handles/Handles.cpp:60:13: note: built-in candidate operator==(volatile struct VkInstance_T *, volatile struct VkInstance_T *)
Vulkan-Hpp/tests/Handles/Handles.cpp:60:13: note: built-in candidate operator==(const volatile struct VkInstance_T *, const volatile struct VkInstance_T *)
Vulkan-Hpp/vulkan/vulkan_handles.hpp:21233:10: note: candidate function (the implicit 'operator==' for this 'operator<=>)'
 21233 |     auto operator<=>( Instance const & ) const = default;
       |          ^
```

</details>

--- a/VulkanHppGenerator.cpp
+++ b/VulkanHppGenerator.cpp
@@ -8250,10 +8250,6 @@ ${typesafeConversionConditionalEnd}
       return m_${memberName} == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( ${className} const & ) const = default;
-#endif
-
   private:
     Vk${className} m_${memberName} = {};
   };

--- a/snippets/HandlesTemplate.hpp
+++ b/snippets/HandlesTemplate.hpp
@@ -14,7 +14,6 @@ namespace VULKAN_HPP_NAMESPACE
   ${handles}
 
   // operators to compare VULKAN_HPP_NAMESPACE::-handles
-#if !defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
   template <typename T, typename std::enable_if<isVulkanHandleType<T>::value, int>::type = 0>
   bool operator==( T const & lhs, T const & rhs )
   {
@@ -50,7 +49,6 @@ namespace VULKAN_HPP_NAMESPACE
   {
     return static_cast<typename T::NativeType>( lhs ) >= static_cast<typename T::NativeType>( rhs );
   }
-#endif
 
   template <typename T, typename std::enable_if<isVulkanHandleType<T>::value, int>::type = 0>
   bool operator==( T const & v, std::nullptr_t )

--- a/vulkan/vulkan_handles.hpp
+++ b/vulkan/vulkan_handles.hpp
@@ -2914,10 +2914,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_surfaceKHR == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( SurfaceKHR const & ) const = default;
-#endif
-
   private:
     VkSurfaceKHR m_surfaceKHR = {};
   };
@@ -3012,10 +3008,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_debugReportCallbackEXT == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( DebugReportCallbackEXT const & ) const = default;
-#endif
 
   private:
     VkDebugReportCallbackEXT m_debugReportCallbackEXT = {};
@@ -3112,10 +3104,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_debugUtilsMessengerEXT == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( DebugUtilsMessengerEXT const & ) const = default;
-#endif
-
   private:
     VkDebugUtilsMessengerEXT m_debugUtilsMessengerEXT = {};
   };
@@ -3201,10 +3189,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_displayKHR == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( DisplayKHR const & ) const = default;
-#endif
 
   private:
     VkDisplayKHR m_displayKHR = {};
@@ -3298,10 +3282,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_swapchainKHR == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( SwapchainKHR const & ) const = default;
-#endif
-
   private:
     VkSwapchainKHR m_swapchainKHR = {};
   };
@@ -3394,10 +3374,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_semaphore == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( Semaphore const & ) const = default;
-#endif
-
   private:
     VkSemaphore m_semaphore = {};
   };
@@ -3489,10 +3465,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_fence == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( Fence const & ) const = default;
-#endif
 
   private:
     VkFence m_fence = {};
@@ -3593,10 +3565,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_performanceConfigurationINTEL == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( PerformanceConfigurationINTEL const & ) const = default;
-#endif
-
   private:
     VkPerformanceConfigurationINTEL m_performanceConfigurationINTEL = {};
   };
@@ -3682,10 +3650,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_queryPool == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( QueryPool const & ) const = default;
-#endif
 
   private:
     VkQueryPool m_queryPool = {};
@@ -3779,10 +3743,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_buffer == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( Buffer const & ) const = default;
-#endif
-
   private:
     VkBuffer m_buffer = {};
   };
@@ -3874,10 +3834,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_pipelineLayout == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( PipelineLayout const & ) const = default;
-#endif
 
   private:
     VkPipelineLayout m_pipelineLayout = {};
@@ -3971,10 +3927,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_descriptorSet == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( DescriptorSet const & ) const = default;
-#endif
-
   private:
     VkDescriptorSet m_descriptorSet = {};
   };
@@ -4066,10 +4018,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_imageView == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( ImageView const & ) const = default;
-#endif
 
   private:
     VkImageView m_imageView = {};
@@ -4163,10 +4111,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_pipeline == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( Pipeline const & ) const = default;
-#endif
-
   private:
     VkPipeline m_pipeline = {};
   };
@@ -4259,10 +4203,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_shaderEXT == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( ShaderEXT const & ) const = default;
-#endif
-
   private:
     VkShaderEXT m_shaderEXT = {};
   };
@@ -4348,10 +4288,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_image == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( Image const & ) const = default;
-#endif
 
   private:
     VkImage m_image = {};
@@ -4449,10 +4385,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_accelerationStructureNV == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( AccelerationStructureNV const & ) const = default;
-#endif
 
   private:
     VkAccelerationStructureNV m_accelerationStructureNV = {};
@@ -4552,10 +4484,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_dataGraphPipelineSessionARM == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( DataGraphPipelineSessionARM const & ) const = default;
-#endif
-
   private:
     VkDataGraphPipelineSessionARM m_dataGraphPipelineSessionARM = {};
   };
@@ -4644,10 +4572,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_opticalFlowSessionNV == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( OpticalFlowSessionNV const & ) const = default;
-#endif
 
   private:
     VkOpticalFlowSessionNV m_opticalFlowSessionNV = {};
@@ -4740,10 +4664,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_descriptorUpdateTemplate == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( DescriptorUpdateTemplate const & ) const = default;
-#endif
 
   private:
     VkDescriptorUpdateTemplate m_descriptorUpdateTemplate = {};
@@ -4838,10 +4758,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_event == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( Event const & ) const = default;
-#endif
 
   private:
     VkEvent m_event = {};
@@ -4941,10 +4857,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_accelerationStructureKHR == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( AccelerationStructureKHR const & ) const = default;
-#endif
-
   private:
     VkAccelerationStructureKHR m_accelerationStructureKHR = {};
   };
@@ -5036,10 +4948,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_micromapEXT == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( MicromapEXT const & ) const = default;
-#endif
 
   private:
     VkMicromapEXT m_micromapEXT = {};
@@ -8137,10 +8045,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_commandBuffer == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( CommandBuffer const & ) const = default;
-#endif
-
   private:
     VkCommandBuffer m_commandBuffer = {};
   };
@@ -8232,10 +8136,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_deviceMemory == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( DeviceMemory const & ) const = default;
-#endif
 
   private:
     VkDeviceMemory m_deviceMemory = {};
@@ -8329,10 +8229,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_videoSessionKHR == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( VideoSessionKHR const & ) const = default;
-#endif
-
   private:
     VkVideoSessionKHR m_videoSessionKHR = {};
   };
@@ -8421,10 +8317,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_deferredOperationKHR == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( DeferredOperationKHR const & ) const = default;
-#endif
 
   private:
     VkDeferredOperationKHR m_deferredOperationKHR = {};
@@ -8517,10 +8409,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_bufferCollectionFUCHSIA == VK_NULL_HANDLE;
     }
-
-#  if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( BufferCollectionFUCHSIA const & ) const = default;
-#  endif
 
   private:
     VkBufferCollectionFUCHSIA m_bufferCollectionFUCHSIA = {};
@@ -8615,10 +8503,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_bufferView == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( BufferView const & ) const = default;
-#endif
-
   private:
     VkBufferView m_bufferView = {};
   };
@@ -8710,10 +8594,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_commandPool == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( CommandPool const & ) const = default;
-#endif
 
   private:
     VkCommandPool m_commandPool = {};
@@ -8807,10 +8687,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_pipelineCache == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( PipelineCache const & ) const = default;
-#endif
-
   private:
     VkPipelineCache m_pipelineCache = {};
   };
@@ -8902,10 +8778,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_cuFunctionNVX == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( CuFunctionNVX const & ) const = default;
-#endif
 
   private:
     VkCuFunctionNVX m_cuFunctionNVX = {};
@@ -8999,10 +8871,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_cuModuleNVX == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( CuModuleNVX const & ) const = default;
-#endif
-
   private:
     VkCuModuleNVX m_cuModuleNVX = {};
   };
@@ -9095,10 +8963,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_cudaFunctionNV == VK_NULL_HANDLE;
     }
-
-#  if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( CudaFunctionNV const & ) const = default;
-#  endif
 
   private:
     VkCudaFunctionNV m_cudaFunctionNV = {};
@@ -9194,10 +9058,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_cudaModuleNV == VK_NULL_HANDLE;
     }
 
-#  if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( CudaModuleNV const & ) const = default;
-#  endif
-
   private:
     VkCudaModuleNV m_cudaModuleNV = {};
   };
@@ -9290,10 +9150,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_descriptorPool == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( DescriptorPool const & ) const = default;
-#endif
 
   private:
     VkDescriptorPool m_descriptorPool = {};
@@ -9389,10 +9245,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_descriptorSetLayout == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( DescriptorSetLayout const & ) const = default;
-#endif
 
   private:
     VkDescriptorSetLayout m_descriptorSetLayout = {};
@@ -9499,10 +9351,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_externalComputeQueueNV == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( ExternalComputeQueueNV const & ) const = default;
-#endif
-
   private:
     VkExternalComputeQueueNV m_externalComputeQueueNV = {};
   };
@@ -9588,10 +9436,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_framebuffer == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( Framebuffer const & ) const = default;
-#endif
 
   private:
     VkFramebuffer m_framebuffer = {};
@@ -9691,10 +9535,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_indirectCommandsLayoutEXT == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( IndirectCommandsLayoutEXT const & ) const = default;
-#endif
-
   private:
     VkIndirectCommandsLayoutEXT m_indirectCommandsLayoutEXT = {};
   };
@@ -9787,10 +9627,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_indirectCommandsLayoutNV == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( IndirectCommandsLayoutNV const & ) const = default;
-#endif
-
   private:
     VkIndirectCommandsLayoutNV m_indirectCommandsLayoutNV = {};
   };
@@ -9882,10 +9718,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_indirectExecutionSetEXT == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( IndirectExecutionSetEXT const & ) const = default;
-#endif
-
   private:
     VkIndirectExecutionSetEXT m_indirectExecutionSetEXT = {};
   };
@@ -9971,10 +9803,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_privateDataSlot == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( PrivateDataSlot const & ) const = default;
-#endif
 
   private:
     VkPrivateDataSlot m_privateDataSlot = {};
@@ -10063,10 +9891,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_renderPass == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( RenderPass const & ) const = default;
-#endif
 
   private:
     VkRenderPass m_renderPass = {};
@@ -10159,10 +9983,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_sampler == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( Sampler const & ) const = default;
-#endif
 
   private:
     VkSampler m_sampler = {};
@@ -10259,10 +10079,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_samplerYcbcrConversion == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( SamplerYcbcrConversion const & ) const = default;
-#endif
-
   private:
     VkSamplerYcbcrConversion m_samplerYcbcrConversion = {};
   };
@@ -10357,10 +10173,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_shaderModule == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( ShaderModule const & ) const = default;
-#endif
-
   private:
     VkShaderModule m_shaderModule = {};
   };
@@ -10453,10 +10265,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_tensorARM == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( TensorARM const & ) const = default;
-#endif
-
   private:
     VkTensorARM m_tensorARM = {};
   };
@@ -10542,10 +10350,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_tensorViewARM == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( TensorViewARM const & ) const = default;
-#endif
 
   private:
     VkTensorViewARM m_tensorViewARM = {};
@@ -10634,10 +10438,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_validationCacheEXT == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( ValidationCacheEXT const & ) const = default;
-#endif
 
   private:
     VkValidationCacheEXT m_validationCacheEXT = {};
@@ -10737,10 +10537,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_videoSessionParametersKHR == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( VideoSessionParametersKHR const & ) const = default;
-#endif
-
   private:
     VkVideoSessionParametersKHR m_videoSessionParametersKHR = {};
   };
@@ -10826,10 +10622,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_pipelineBinaryKHR == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( PipelineBinaryKHR const & ) const = default;
-#endif
 
   private:
     VkPipelineBinaryKHR m_pipelineBinaryKHR = {};
@@ -11103,10 +10895,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_queue == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( Queue const & ) const = default;
-#endif
 
   private:
     VkQueue m_queue = {};
@@ -18358,10 +18146,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_device == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( Device const & ) const = default;
-#endif
-
   private:
     VkDevice m_device = {};
   };
@@ -18453,10 +18237,6 @@ namespace VULKAN_HPP_NAMESPACE
     {
       return m_displayModeKHR == VK_NULL_HANDLE;
     }
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( DisplayModeKHR const & ) const = default;
-#endif
 
   private:
     VkDisplayModeKHR m_displayModeKHR = {};
@@ -20417,10 +20197,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_physicalDevice == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( PhysicalDevice const & ) const = default;
-#endif
-
   private:
     VkPhysicalDevice m_physicalDevice = {};
   };
@@ -21229,10 +21005,6 @@ namespace VULKAN_HPP_NAMESPACE
       return m_instance == VK_NULL_HANDLE;
     }
 
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
-    auto operator<=>( Instance const & ) const = default;
-#endif
-
   private:
     VkInstance m_instance = {};
   };
@@ -21351,7 +21123,6 @@ namespace VULKAN_HPP_NAMESPACE
 #endif /* VULKAN_HPP_DISABLE_ENHANCED_MODE */
 
   // operators to compare VULKAN_HPP_NAMESPACE::-handles
-#if !defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
   template <typename T, typename std::enable_if<isVulkanHandleType<T>::value, int>::type = 0>
   bool operator==( T const & lhs, T const & rhs )
   {
@@ -21387,7 +21158,6 @@ namespace VULKAN_HPP_NAMESPACE
   {
     return static_cast<typename T::NativeType>( lhs ) >= static_cast<typename T::NativeType>( rhs );
   }
-#endif
 
   template <typename T, typename std::enable_if<isVulkanHandleType<T>::value, int>::type = 0>
   bool operator==( T const & v, std::nullptr_t )

